src/entities/trader/base-trader.ts

```diff
@@ -215,47 +254,55 @@ export class Trader {
   }
 
   public async exerciseOption({
-    optionId,
+    option,
     amount,
-    clearinghouse,
   }: {
-    optionId: bigint;
+    option: Option;
     amount: bigint;
-    clearinghouse: ClearinghouseContract;
   }) {
-    // prepare tx
-    const { request } = await clearinghouse.simulate.exercise([
-      optionId,
+    await option.exerciseOption({
       amount,
-    ]);
-    // send tx
-    const receipt = await this.executeTransaction(
-      request as SimulatedTxRequest,
-    );
-    // check result
-    if (receipt.status === 'success') {
-      console.log(
-        `Successfully exercised ${amount}x options, with ID ${optionId.toString()}`,
-      );
-    }
+      trader: this,
+    });
   }
 
-  public async redeemClaim({
-    optionId,
-    clearinghouse,
-  }: {
-    optionId: bigint;
-    clearinghouse: ClearinghouseContract;
-  }) {
-    // prepare tx
-    const { request } = await clearinghouse.simulate.redeem([optionId]);
-    // send tx
-    const receipt = await this.executeTransaction(
-      request as SimulatedTxRequest,
-    );
-    // check result
-    if (receipt.status === 'success') {
-      console.log(`Successfully redeemed claim with ID ${optionId.toString()}`);
+  public async redeemClaim({ claim }: { claim: Claim }) {
+    await claim.redeemClaim(this);
+  }
```

Comment:
```suggestion
  public redeemClaim({ claim }: { claim: Claim }) {
    return claim.redeemClaim(this);
  }
```
Useless async/await.

Reply:
I removed this and exercise option from the trader class, seeing as we now have the claim/option entities going